tests/4_functions.cpp

```diff
@@ -0,0 +1,81 @@
+#include <cmath>
+
+#include "catch.hpp"
+#include "calculate.h"
+
+#define eval(e) calculate::Calculate(e)()
+#define approx(x) Approx(x).epsilon(1e-12)
+
+
+TEST_CASE("Builtin functions", "[functions]") {
+
+    SECTION("Other functions") {
+        CHECK(eval("fabs(3.1416)") == approx(3.1416));
+        CHECK(eval("abs(-1)") == approx(1));
+        CHECK(eval("fma(10, 20, 30)") == approx(230));
+        CHECK(eval("copysign(10, -1)") == approx(-10));
+        CHECK(eval("nextafter(0.0, 1.0)") == approx(4.94065645841e-324));
+    }
+
+    SECTION("Minimum, maximum and difference functions") {
+        CHECK(eval("fdim(2, 1)") == approx(1));
+        CHECK(eval("fmax(-100, 1)") == approx(1));
+        CHECK(eval("fmin(-100, -1)") == approx(-100));
+    }
+
+    SECTION("Rounding and remainder functions") {
+        CHECK(eval("ceil(2.3)") == approx(3));
+        CHECK(eval("floor(3.8)") == approx(3));
+        CHECK(eval("fmod(5.3, 2)") == approx(1.3));
+        CHECK(eval("trunc(5.5)") == approx(5));
+        CHECK(eval("round(5.5)") == approx(6));
+        CHECK(eval("rint(-3.8)") == approx(-4));
+        CHECK(eval("nearbyint(-2.3)") == approx(-2));
+        CHECK(eval("remainder(18.5, 4.2)") == approx(1.7));
+    }
+
+    SECTION("Power functions") {
+        CHECK(eval("pow(32.01, 1.54)") == approx(208.036691405387));
+        CHECK(eval("sqrt(1024)") == approx(32));
+        CHECK(eval("cbrt(27)") == approx(3));
+        CHECK(eval("hypot(3, 4)") == approx(5));
+    }
+
+    SECTION("Exponential and logarithmic functions") {
+        CHECK(eval("exp(5)") == approx(148.413159102577));
```

Comment:
All the examples for the builtin functions tests have been extracted from [www.cplusplus.com](http://www.cplusplus.com/reference/cmath/).